plutus-pab/src/Plutus/PAB/Core/ContractInstance.hs

```diff
@@ -148,18 +148,18 @@ processAwaitSlotRequestsSTM =
     maybeToHandler (extract Contract.Effects._AwaitSlotReq)
     >>> (RequestHandler $ \targetSlot_ -> fmap AwaitSlotResp . InstanceState.awaitSlot targetSlot_ <$> ask)
 
-processTxConfirmedRequestsSTM ::
+processTxStatusChangeRequestsSTM ::
     forall effs.
     ( Member (Reader BlockchainEnv) effs
     )
     => RequestHandler effs PABReq (STM PABResp)
-processTxConfirmedRequestsSTM =
-    maybeToHandler (extract Contract.Effects._AwaitTxConfirmedReq)
+processTxStatusChangeRequestsSTM =
+    maybeToHandler (extract Contract.Effects._AwaitTxStatusChangeReq)
     >>> RequestHandler handler
     where
-        handler req = do
+        handler txId = do
             env <- ask
-            pure (AwaitTxConfirmedResp . unTxConfirmed <$> InstanceState.waitForTxConfirmed req env)
+            pure (AwaitTxStatusChangeResp txId <$> InstanceState.waitForTxStatusChange Unknown txId env)
```

Comment:
Why is the `oldStatus` hardcoded to `Unknown`? Don't we need to fetch the `oldStatus` of `txId`?

Reply:
Getting the old status of `txId` is tricky - we don't want to store the ID of every single transaction that has been validated forever, because then we would need unbounded space (proportional to the size of the chain). So in the PAB we only store the transaction IDs of the last `k` blocks, where `k` is a fixed constant, the maximum number of blocks that can be rolled back